### PR TITLE
Added linked_list_middle_element.cpp which detects the middle element…

### PR DESCRIPTION
**Pull request by Sarthak-9**
… of a linked list and updated README.md

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

Algorithm to find the middle element of a Linked List

### Checklist

- [#] I've read the contribution guidelines.
- [#] I've checked the issue list before deciding what to submit.
- [#] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

#1617

### CODE REVIEW

**Comment by Sarthak-9** [comment]
@shraddhavp Please review

File: C-Plus-Plus/ds/linked_list_middle_element.cpp [hunk 1]
```diff
@@ -0,0 +1,122 @@
+/*
+
+ Created by Sarthak-9
```
Comment (HarshCasper) [inline C-Plus-Plus/ds/linked_list_middle_element.cpp]: If you would like to be cited for the Source Code, please add `Author: Sarthak Saxea (@Sarthak-9)`

File: C-Plus-Plus/ds/linked_list_middle_element.cpp [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+ Enter the size of list
+ 8
+ Enter the elements of linked list
+ 45 24 32 72 54 18 90 65
+
+ Output :
+ Middle Element (Method-I): 72
+ Middle Element (Method-II): 72
+
+*/
```
Comment (HarshCasper) [inline C-Plus-Plus/ds/linked_list_middle_element.cpp]: Add a new line at the EOF

File: C-Plus-Plus/ds/linked_list_middle_element.cpp [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+#include <cmath>
+#include <stack>
+
+using namespace std;
+
+class Node{
+        public:
+                int data;
+                Node* next;
+};
```
Comment (HarshCasper) [inline C-Plus-Plus/ds/linked_list_middle_element.cpp]: What level of indentation are you following? This is incoherent with the code that exists in the repo.

File: C-Plus-Plus/ds/linked_list_middle_element.cpp [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+void create(int arr[], int n){
+        Node* temp;
+        Node* tail;
+
+        head->data = arr[0];
+        head->next = nullptr;
+        tail = head;
+
+        // creates the linked list from array
+        for (int i=1; i<n; i++){
```
Comment (HarshCasper) [inline C-Plus-Plus/ds/linked_list_middle_element.cpp]: Format the Code

**Review by HarshCasper — CHANGES_REQUESTED** [review]
Remove unused lines of Code. Changes suggested are tiny ones. 

I will let @iamrajiv and @atarax665 to take a look at the Code. From my side, it looks good.

**Comment by Sarthak-9** [comment]
@HarshCasper Check please

File: C-Plus-Plus/ds/linked_list_middle_element.cpp [hunk 5]
```diff
@@ -0,0 +1,118 @@
+/*
+ Author: Sarthak Saxena (@Sarthak-9)
+
+ A Linked List is a linear data structure which includes a series of connected nodes. Each node stores the data and the address of the next node.
+ Middle Element can be extracted using following two methods :
+ Time complexity of MiddleNode1 : O(n)
```
Comment (atarax665) [inline C-Plus-Plus/ds/linked_list_middle_element.cpp:7]: Please do add a short description of both the methods individually.

File: C-Plus-Plus/ds/linked_list_middle_element.cpp [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+
+        int length = 0;
+
+        // counts the length
+        while (p){
+                length++;
+                p = p->next;
+        }
+
+        int index = (int)ceil(length/2.0);
```
Comment (atarax665) [inline C-Plus-Plus/ds/linked_list_middle_element.cpp]:
```suggestion
        int index = (int) ceil(length / 2.0);
```

File: C-Plus-Plus/ds/linked_list_middle_element.cpp [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+
+        Node* q = head;
+
+        // traverses to the middle element
+        for (int i=0; i<index-1; i++){
```
Comment (atarax665) [inline C-Plus-Plus/ds/linked_list_middle_element.cpp]:
```suggestion
        for (int i = 0; i < index - 1; i++){
```

**Review by atarax665 — CHANGES_REQUESTED** [review]

**Comment by Sarthak-9** [comment]
@atarax665 Check now

**Review by atarax665 — APPROVED** [review]
LGTM :+1:

**Review by iamrajiv — APPROVED** [review]

**Pull request merged**